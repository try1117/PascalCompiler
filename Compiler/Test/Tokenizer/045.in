var
  a:integer = $fff;
  b:double = -0.028e+7;
begin
  //if a = 2 * b
  if a = b then
    write(3 * a);
end.
