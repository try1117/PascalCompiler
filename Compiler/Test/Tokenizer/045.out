1  | 1  | KEYWORD_VAR              | var                      | 
2  | 3  | VARIABLE                 | a                        | 
2  | 4  | OP_COLON                 | :                        | 
2  | 5  | KEYWORD_INTEGER          | integer                  | 
2  | 13 | OP_EQUAL                 | =                        | 
2  | 15 | CONST_HEX                | fff                      | 4095
2  | 19 | SEP_SEMICOLON            | ;                        | 
3  | 3  | VARIABLE                 | b                        | 
3  | 4  | OP_COLON                 | :                        | 
3  | 5  | KEYWORD_DOUBLE           | double                   | 
3  | 12 | OP_EQUAL                 | =                        | 
3  | 14 | OP_MINUS                 | -                        | 
3  | 15 | CONST_DOUBLE             | 0.028e+7                 | 280000.000000000000000
3  | 23 | SEP_SEMICOLON            | ;                        | 
4  | 1  | KEYWORD_BEGIN            | begin                    | 
6  | 3  | KEYWORD_IF               | if                       | 
6  | 6  | VARIABLE                 | a                        | 
6  | 8  | OP_EQUAL                 | =                        | 
6  | 10 | VARIABLE                 | b                        | 
6  | 12 | KEYWORD_THEN             | then                     | 
7  | 5  | KEYWORD_WRITE            | write                    | 
7  | 10 | SEP_BRACKET_LEFT         | (                        | 
7  | 11 | CONST_INTEGER            | 3                        | 3
7  | 13 | OP_MULT                  | *                        | 
7  | 15 | VARIABLE                 | a                        | 
7  | 16 | SEP_BRACKET_RIGHT        | )                        | 
7  | 17 | SEP_SEMICOLON            | ;                        | 
8  | 1  | KEYWORD_END              | end                      | 
8  | 4  | SEP_DOT                  | .                        | 
